elki-core-math/src/main/java/elki/math/statistics/tests/mcde/MWPTest.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * This file is part of ELKI:
+ * Environment for Developing KDD-Applications Supported by Index-Structures
+ *
+ * Copyright (C) 2019
+ * ELKI Development Team
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package elki.math.statistics.tests.mcde;
+
+import elki.utilities.datastructures.arraylike.NumberArrayAdapter;
+import elki.utilities.documentation.Reference;
+import elki.utilities.exceptions.AbortException;
+import elki.utilities.random.RandomFactory;
+
+import java.lang.reflect.Array;
+
+import static elki.math.statistics.distribution.NormalDistribution.erf;
+
+/**
+ * Implementation of Mann-Whitney U test returning the p-value (not the test
+ * statistic, thus MWP) for MCDEDependenceMeasure. Implements algorithm 1 and 3
+ * of reference paper.
+ *
+ * @author Alan Mazankiewicz
+ * @author Edouard Fouché
+ */
+
+@Reference(authors = "Edouard Fouché, Klemens Böhm", //
```

Comment:
Because `@Reference` annotates the `class`, there shouldn't be an empty line inbetween.